### PR TITLE
Scaffolding for react-dom/unstable_external-server-runtime

### PR DESCRIPTION
**Pull request by acdlite**
Alternative to #25459.

Implements a new bundle type for Rollup called BROWSER_SCRIPT. This is intended for scripts that get delivered straight to the browser without needing to be processed by a bundler.

Right now there's only a single use case so I didn't stress about making it general purpose.

The use case is: a script that loads the Fizz browser runtime, and sets up a MutationObserver to receive instructions as HTML streams in. This will be an alternative option to the default Fizz behavior of sending the runtime down as inline script tags, to accommodate environments where inline script tags are not allowed.

There's no development version of this bundle because it doesn't contain any warnings or run any user code.

None of the actual implementation is in this PR; it just sets up the build infra.

[Prettified view of resulting file](https://react-builds.vercel.app/commits/6c1485e8bfdfb17d067aaf4e8898289d6b75c4d1/files/oss-experimental/react-dom/unstable_server-external-runtime.js?pretty=true)

### CODE REVIEW

**Comment by sebmarkbage** [comment]
`"use strict"` seems unnecessary. We should be able to set `output: { strict: false }` on rollup.

**Comment by acdlite** [comment]
Annoyingly that did not work. I think it might be added by Babel or something.

**Review by sebmarkbage — COMMENTED** [review]
We probably shouldn't even run Babel on this. The idea was to let closure handle everything longer term anyway. It handles most of it already, and we should probably just avoid those other syntaxes.

**Review by sebmarkbage — APPROVED** [review]

**Comment by acdlite** [comment]
Turns out it was Closure. Fixed by setting output format to ES5. Maybe we should do that for all modules? (The default is `ECMASCRIPT_NEXT`, whatever that means.) I left a TODO.

**Comment by sebmarkbage** [comment]
It's already set to `ECMASCRIPT5_STRICT` so we shouldn't override that for other bundles. That's probably why you got strict mode though.

**Comment by acdlite** [comment]
Oh nvm, I didn't see that! Yeah I'll set that in the else condition.

**Comment by acdlite** [comment]
Or I guess refactor this object assign thingy so it doesn't set it at all

**Comment by sebmarkbage** [comment]
Curious to see what happens when these builds complete though.

**Comment by acdlite** [comment]
> Curious to see what happens when these builds complete though.

Oh sorry I pushed before I saw this. I can open a separate PR for that.

**Comment by sebmarkbage** [comment]
I looked already. It got larger. Mostly `var` -> `const` because we transpile most other things away anyway but `const` is significant for closure so we intentionally don't.

File: scripts/rollup/build.js [hunk 1]
```diff
@@ -371,14 +363,24 @@ function getPlugins(
...
     isProduction &&
-      closure(
-        Object.assign({}, closureOptions, {
-          // Don't let it create global variables in the browser.
-          // https://github.com/facebook/react/issues/10909
-          assume_function_wrapper: !isUMDBundle,
-          renaming: !shouldStayReadable,
-        })
-      ),
+      closure({
```
Comment (sebmarkbage) [inline scripts/rollup/build.js:366]: ✅ No abstraction.

File: scripts/rollup/build.js [hunk 2]
```diff
@@ -371,14 +363,24 @@ function getPlugins(
     isUMDBundle && entry === 'react-art' && commonjs(),
     // Apply dead code elimination and/or minification.
     isProduction &&
-      closure(
-        Object.assign({}, closureOptions, {
```
Comment (sebmarkbage) [inline scripts/rollup/build.js:375]: 🤬 Abstraction.

**Comment by acdlite** [comment]
Ok it looks good now: https://react-builds.vercel.app/commits/6c1485e8bfdfb17d067aaf4e8898289d6b75c4d1/files/oss-experimental/react-dom/unstable_server-external-runtime.js?pretty=true

**Pull request merged**